src/formatDate.js

```diff
@@ -8,7 +8,60 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
-}
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const toSeparator = toFormat[toFormat.length - 1];
+
+  const dateParts = date.split(fromSeparator);
+  const newDate = [];
+  const newDateObj = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (
+      fromFormat[i] === 'YYYY' ||
+      fromFormat[i] === 'YY' ||
+      fromFormat[i] === 'MM' ||
+      fromFormat[i] === 'DD'
+    ) {
+      newDateObj[fromFormat[i]] = dateParts[i];
+    }
```

Comment:
The conditional check inside the loop seems to be unnecessary since the `fromFormat` array is expected to contain only valid date parts as per task instructions. You could directly assign the values without checking them.